### ISSUE TITLE
Manticore leaves lots of files in /tmp

### ISSUE THREAD

**Issue by offlinemark** [trailofbits/manticore#568]
When the analyzed program called `open` and the first byte of the filename is symbolic, Manticore creates a /tmp file, fills it with 64 '+' and uses that as the backing file.

- Could manticore clean up after itself? this can probably be done using `tempfile.TemporaryFile` rather than `tempfile.mkstemp` in linux.py:SLinux.sys_open
- Does manticore need to have a file on disk at all? could the symbolic file exist only in memory?

**Comment by MrSynAckSter** [comment]
I changed mkstmp to tempfile.TemporaryFile in https://github.com/trailofbits/manticore/commit/221f1493bb0845625714144743115c605b6235b1. I ran it against all of the tests in /tests/native, and there didn't seem to be errors. My only concern is that if tempfile.TemporaryFile is ever garbage collected, the file cease to exist. I don't know enough about how Manticore handles these files to say whether or not that possible, but running it against a few random binaries that use sys_open doesn't seem to cause any issues of that nature. If you guys can suggest anything else to do in order to ensure this change is safe, let me know.
